packages/browser/src/Session.ts

```diff
@@ -227,6 +227,31 @@ export class Session extends EventEmitter {
         this
       )
     );
+
+    // When a session is logged in, we want to track its ID in local storage to
+    // enable silent refresh.
+    this.on(EVENTS.LOGIN, () => {
+      // Store the current session ID specifically in 'localStorage' (i.e., not using
+      // any other storage mechanism), as we don't deem this information to be
+      // sensitive, and we want to ensure it survives a browser tab refresh.
+      window.localStorage.setItem(KEY_CURRENT_SESSION, this.info.sessionId);
+    });
+
+    // On logout, silent refresh should no longer be considered a viable option.
+    this.on(EVENTS.LOGOUT, () => {
+      window.localStorage.removeItem(KEY_CURRENT_SESSION);
+    });
+
+    this.on(EVENTS.SESSION_EXPIRED, () => {
+      window.localStorage.removeItem(KEY_CURRENT_SESSION);
```

Comment:
Does this also need to `await this.logout()` or is error emitted afterwards?

Reply:
Fixed in a66d20282c4d4abab963a65a6449e6e364ede7ba, the error wasn't related to an actual logout indeed.